src/core/programs.rkt

```diff
@@ -116,39 +129,67 @@
     [(? literal?) '()]
     [(? number?) '()]
     [(? variable?) (list prog)]
-    [`(,op ,args ...) (remove-duplicates (append-map free-variables args))]))
+    [(approx spec impl) (remove-duplicates (append (free-variables spec) (free-variables impl)))]
+    [(list _ args ...) (remove-duplicates (append-map free-variables args))]))
 
 (define (replace-vars dict expr)
```

Comment:
Are we sure this is never called with express instead of vars? Is this the fun only ever used by preprocessing?

Reply:
Seems to be used in a few places. I think for `approx` it should only check `impl` through.